### PR TITLE
chore: Update README

### PR DESCRIPTION
**Pull request by fardin01**
This commit updates the README to include the findings in https://github.com/corazawaf/coraza-caddy/issues/137, because it is useful to have this information directly in the README rather than having to search for it.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -125,3 +125,27 @@ go run mage.go buildCaddy
...
+ @block_codes `{err.status_code} in [403]`
+ handle @block_codes {
+  root    * /path/to/html/dir
+  rewrite * /{err.status_code}.html
+  file_server
+ }
+}
+```
+
+It is possible to use the [templates](https://caddyserver.com/docs/caddyfile/directives/templates) directive to render data dynamically. Take a look at the `403.html` file in the `example` folder.
```
Comment (M4tteoP) [inline README.md:151]:
```suggestion
It is possible to use the [templates](https://caddyserver.com/docs/caddyfile/directives/templates) directive to render data dynamically. Take a look at [`example/403.html`](./example/403.html) file. 
```
nit

File: example/403.html [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      h1 {
+        font-size: 5em;
+        margin: 0;
+      }
+    </style>
+  </head>
+  <body>
+    <div class="access-denied-container">
+      <h1>Access Denied</h1>
+      <p>Host: {{.Host}}</p>
```
Comment (jptosso) [inline example/403.html:35]: maybe its a dumb question, but no chances of XSS here?

Comment (fardin01) [reply example/403.html:35]: That's a good point. This is just to demonstrate the templating capability. How would you suggest demonstrating the templating feature?

Comment (fzipi) [reply example/403.html:35]: Maybe use `stripHTML`?
```
stripHTML

Removes HTML from a string.

{{stripHTML "Shows <b>only</b> text content"}}
```

Comment (fardin01) [reply example/403.html]: Done.

**Review by fzipi — APPROVED** [review]
LGTM. The next obvious step will be (in a followup PR) to add this example as part of the github action build process, so we can test everything works properly 😄

**Comment by fardin01** [comment]
> LGTM. The next obvious step will be (in a followup PR) to add this example as part of the github action build process, so we can test everything works properly 😄

Great. 

Do you mean adding it to e2e tests?

**Comment by fzipi** [comment]
🤔 It was more about checking that the example files are sane and won't break anything if used.  
I was thinking on a simple GHA that starts a caddy server with the example files and we test the response using curl or similar to see if it contains the host, etc. But if that is about e2e, then could be.

**Comment by M4tteoP** [comment]
Note: tests are failing because of https://github.com/corazawaf/coraza-caddy/pull/165, this branch needs to be rebased once that PR is merged

**Pull request merged**